### PR TITLE
Kafka node new client

### PR DESCRIPTION
**Pull request by amiram**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/SOHU-Co/kafka-node>
- [X] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/kafka-node/index.d.ts [hunk 1]
```diff
@@ -1,99 +1,126 @@
...
+}
+
+export interface ProducerOptions {
+  requireAcks?: number;
+  ackTimeoutMs?: number;
+  partitionerType?: number;
+}
+
+export interface KafkaClientOptions {
+    kafkaHost?: string;
```
Comment (sfrooster) [inline types/kafka-node/index.d.ts]: Add "clientId?: string;"?

File: types/kafka-node/index.d.ts [hunk 2]
```diff
@@ -1,99 +1,126 @@
...
-    constructor(client: Client, fetchRequests: Array<OffsetFetchRequest>, options: ConsumerOptions);
+export class Consumer {
+    constructor(client: Client, fetchRequests: Array<OffsetFetchRequest | string>, options: ConsumerOptions);
     client: Client;
-    on(eventName: string, cb: (message: string) => any): void;
-    on(eventName: string, cb: (error: any) => any): void;
-    addTopics(topics: Array<string>, cb: (error: any, added: boolean) => any): void;
-    addTopics(topics: Array<Topic>, cb: (error: any, added: boolean) => any, fromOffset: boolean): void;
-    removeTopics(topics: Array<string>, cb: (error: any, removed: boolean) => any): void;
+    on(eventName: "message", cb: (message: Message) => any): void;
```
Comment (sfrooster) [inline types/kafka-node/index.d.ts]: "message" | "done"

**Review by sfrooster — COMMENTED** [review]
Also wonder if the following should be supported:
on("connect"
on("close"
on("reconnect"
on("brokersChanged"
on("zkReconnect"
on("rebalancing"
on("rebalanced"
on("socker_error"

File: types/kafka-node/index.d.ts (hunk 2 above)
Comment (amiram) [inline types/kafka-node/index.d.ts]: I've added a type to the message param in the callback. I'm not sure it is correct for the done event

**Comment by amiram** [comment]
About all the other events, they are not documented anywhere and I'm not sure that we should add them.

File: types/kafka-node/index.d.ts (hunk 1 above)
Comment (amiram) [inline types/kafka-node/index.d.ts]: done

**Review by sfrooster — APPROVED** [review]

**Pull request merged**